builder/build_test.go

```diff
@@ -287,20 +287,49 @@ func Test_buildFlagSlice(t *testing.T) {
 				t.Errorf("Slices differ in size - wanted: %d, found %d", len(test.expectedSlice), len(flagSlice))
 			}
 
-			//map iteration is non-deterministic so dont compare values if the buildArgMap is > 1
-			if len(test.buildArgMap) <= 1 {
-				for i, val := range flagSlice {
+			isMatch := compareSliceValues(test.expectedSlice, flagSlice)
+			if !isMatch {
+				t.Errorf("Slices differ in values - wanted: %v, found %s", test.expectedSlice, flagSlice)
+			}
 
-					if val != test.expectedSlice[i] {
-						t.Errorf("Slices differ in values - wanted: %s, found %s", test.expectedSlice[i], val)
-					}
+		})
+	}
 
-				}
+}
+
+func compareSliceValues(expectedSlice, outputSlice []string) bool {
+	var expectedValueMap = make(map[string]int)
+
+	for _, expectedValue := range expectedSlice {
+		if _, ok := expectedValueMap[expectedValue]; ok {
+			expectedValueMap[expectedValue]++
+		} else {
+			expectedValueMap[expectedValue] = 1
+		}
+	}
+
+	var outputValueMap = make(map[string]int)
+	for _, outputValue := range outputSlice {
+		if _, ok := outputValueMap[outputValue]; ok {
+			outputValueMap[outputValue]++
+		} else {
+			outputValueMap[outputValue] = 1
+		}
+	}
+
+	for key, expectedValue := range expectedValueMap {
+
+		if value, ok := expectedValueMap[key]; ok {
```

Comment:
Should this use the `outputValueMap` instead? 

```
		if value, ok := outputValueMap[key]; ok {
```

You iterate over the `expectedValueMap` and then select again form `expectedValueMap`, it seems tautological.


You could also reduce the amount of nesting and make the code a little more symantec by doing something like this 

```go
for key, expectedValue := range expectedValueMap {
	value, exists := outputValueMap[key]
	if !exists || value != expectedValue  {
		return false
	}
}
```
I think this if statement reads closer to how your would describe the code, "if the value does not exist or is not the expected value, return false"  The order becomes important in the if statement, because you only want to check the value after you know the value exists.


I think you need to include a length check to ensure that there are no extra values in the output map. This should probably be the first thing that happens.  

Alternatively, this can also be verified while you construct the `outputValueMap`. If you find a value that does not exist in the `expectedValueMap`, then you can return before you even check the check the values in this loop. 


Lastly, the role of the values might be clearer if we use `actual*` instead of `output*`.   This will align closer to the error messages we use in other tests.

Reply:
Thanks for the feedback. Yes that should be `outputValueMap '. 

I have updated with all changes from review comments.